README.md

```diff
@@ -1,2 +1,310 @@
-# validator
-A tiny runtime library for type assertions
+# Validator
+
+A tiny library for run time type assertions.
```

Comment:
I think this it might be worthwhile to explain a little more about the use cases, the benefits, and especially why existing libraries didn't suit our needs.